live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java

```diff
@@ -29,64 +29,81 @@
 
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.List;
 import java.util.Map;
 
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.camel.CamelContext;
 import org.apache.camel.Exchange;
 import org.apache.camel.LoggingLevel;
 import org.apache.camel.builder.RouteBuilder;
 import org.apache.camel.model.dataformat.JsonLibrary;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
-import org.alfresco.hxi_connector.live_ingester.domain.exception.LiveIngesterRuntimeException;
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationConfig;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointClientErrorException;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointServerErrorException;
 
 @Component
+@RequiredArgsConstructor
 @Slf4j
 public class PreSignedUrlRequester extends RouteBuilder implements StorageLocationRequester
 {
     private static final String LOCAL_ENDPOINT = "direct:" + PreSignedUrlRequester.class.getSimpleName();
+    private static final String LOCAL_RETRYABLE_ENDPOINT = "direct:retryable-" + PreSignedUrlRequester.class.getSimpleName();
     static final String ROUTE_ID = PreSignedUrlRequester.class.getSimpleName();
+    private static final String UNEXPECTED_STATUS_CODE_MESSAGE = "Unexpected response status code - expecting: %d, received: %d";
+    private static final int EXPECTED_STATUS_CODE = 201;
     static final String STORAGE_LOCATION_PROPERTY = "preSignedUrl";
     static final String NODE_ID_PROPERTY = "objectId";
     static final String CONTENT_TYPE_PROPERTY = "contentType";
-    private static final int EXPECTED_STATUS_CODE = 201;
 
     private final CamelContext camelContext;
-
-    private final String targetEndpoint;
-
-    @Autowired
-    public PreSignedUrlRequester(CamelContext camelContext, @Value("${alfresco.integration.storage.endpoint}") String targetEndpoint)
-    {
-        super(camelContext);
-        this.camelContext = camelContext;
-        this.targetEndpoint = targetEndpoint;
-    }
+    private final IntegrationConfig.Properties integrationProperties;
 
     @Override
+    @SuppressWarnings("unchecked")
     public void configure()
     {
-        onException(Exception.class)
-                .log(LoggingLevel.ERROR, log, "Unexpected response. Body: ${body}")
+        List<Class<? extends Throwable>> retryReasons = integrationProperties.storage().retry().reasons();
+        // @formatter:off
+        if (retryReasons != null && !retryReasons.isEmpty())
+        {
+            onException(retryReasons.toArray(Class[]::new))
```

Comment:
Presumably this applies to both routes defined below, in which case what is the point of having an intermediate `LOCAL_RETRYABLE_ENDPOINT`?

Reply:
+1